### ISSUE TITLE
Cannot connect to docker.for.mac.localhost

### ISSUE THREAD

**Issue by alxlu** [SeleniumHQ/docker-selenium#511]
## Meta -
Image(s):  
node-chrome(-debug)
Docker-Selenium Image Version(s):  
latest

OS: 
macOS

## Expected Behavior -
Navigating to `docker.for.mac.localhost:PORT` on Google Chrome [will resolve to the internal IP used by the host](https://docs.docker.com/docker-for-mac/networking/).

## Actual Behavior -
`docker.for.mac.localhost refused to connect.`
If I connect to the container with VNC and run wget docker.for.mac.localhost:8080, it's able to successfully resolve the IP and download the file. It also seems to be working properly on node-firefox.

**Comment by ddavison** [comment]
could we see the command you are using / script that you're using to run against that uri?

**Comment by alxlu** [comment]
Sure! I'm running my tests with `webdriver.io`, and just navigating to it using
```javascript
browser.url(`${URL}/upload`);
// URL is passed in as an env variable--in this case pointing to
// http://docker.for.mac.localhost:8080
```

Also, the issue can be reproduced by using the config below, VNCing into them and then manually navigating to the URL.

`docker-compose.yml`
```yaml
selenium-hub:
  image: selenium/hub
  ports:
    - "4444:4444"
chrome:
    image: selenium/node-chrome-debug
    ports:
      - "9000:5900"
    links:
      - selenium-hub:hub
firefox:
    image: selenium/node-firefox-debug
    ports:
      - "9001:5900"
    links:
      - selenium-hub:hub
```
You can start any kind local web server on the host (e.g. `python -m SimpleHTTPServer` in any directory, then VNC into both the Firefox and Chrome containers and launch their respective browsers and manually navigate them to `http://docker.for.mac.localhost:8000`

Firefox navigates to it properly, but Chrome does not

![screen shot 2017-06-30 at 2 57 32 pm](https://user-images.githubusercontent.com/3271834/27755435-87272032-5da4-11e7-86ed-2aeadbb1c944.png)

**Comment by ddavison** [comment]
i'm not a javascript guy, but could it be the use of the grave?

in most languages,

```
'' != ""
"" != ``
```

have you tried with single/double quotes?

**Comment by alxlu** [comment]
That's a template literal, but the issue isn't with JS.

Sorry if I was a little unclear&mdash;`8080` is from my project, but I figured it would be easier to reproduce the problem by spinning up the docker containers, and creating any kind of local web server and manually navigating to the address. The screenshots below from running `python -m SimpleHTTPServer` which defaults to `8000`.

**Comment by ddavison** [comment]
hrm.. ah no problem :)  just trying to eliminate the obvious!

> since it can't navigate there even if you VNC in and manually enter the address.

I'm under the assumption based on your issue, you meant

> since it CAN navigate there even if you VNC in and manually enter the address.

if this is the case, then i cannot imagine what's going wrong here.  i'll be flying home tomorrow, so probably won't be able to get back to this issue in a while; hopefully @diemol can chime in here.

**Comment by alxlu** [comment]
Oh, I meant Chrome **can't** navigate there even if you VNC in there manually. However, Firefox **can**. Also, other tools within the Chrome container are able to such as `wget`
![screen shot 2017-06-30 at 3 31 30 pm](https://user-images.githubusercontent.com/3271834/27756070-36a1a2cc-5da9-11e7-9621-9f0f35dcfb1c.png)

**Comment by diemol** [comment]
Sorry to jump in so late. I ran into the same issue recently, and I am asking about it here https://github.com/moby/moby/issues/22753

I am assuming that it is entirely an issue related to Chrome. But we need to check if we need to adjust something in the config for Chrome, somehow I guess it is not getting this DNS entry.

**Comment by diemol** [comment]
Hi @alxlu, 
After Googling for a while and checking different links, this issue explains it briefly: https://github.com/docker/for-mac/issues/1837

So, it is an issue between docker and Chrome, since Chrome is not going to resolve the hostname.

As a workaround, I suggest to use the IP `192.168.65.1` instead of the `docker.for.mac.localhost`.  Also to follow up on the issue to see how it ends up.

Hopefully docker will change the name, since `localhost` should be only for local traffic.

I'll close this issue since it is not related to docker-selenium.

**Comment by jedwards1211** [comment]
@diemol I tried `192.168.65.1` and Chrome complains that "There is no internet connection" even though Firefox connects to that IP just fine.

**Comment by jedwards1211** [comment]
Nevermind, I forgot that a program I'm running called Freedom messes up networking for some Docker images.

**Comment by u007** [comment]
i have the same issue, firefox works, but not chrome.
i did not have any kind of custom network except running local dns that resolved to cloudflare dns 1.1.1.1
